### PR TITLE
Initial columns functions

### LINKED ISSUE #9: columns argument (plus joins)
### Description of feature

Getting some columns requires doing a join. For example, getting [protein info for transcripts from  `ensembldb`](https://bioconductor.org/packages/release/bioc/vignettes/ensembldb/inst/doc/proteins.html#2_Fetch_protein_annotation_for_genes_and_transcripts):

```r
transcripts(edb, filter = GeneNameFilter("ZBTB16"),
           columns = c("protein_id", "uniprot_id", "tx_biotype"))
```

We should be able to do this as well. This requires being able to figure out which table a column is in, then joining on that column.

### PR DESCRIPTION
**Pull request by lauradmartens**
Fixes #9 when finished and partly #6 for discoverability of columns

### CODE REVIEW

**Comment by lauradmartens** [comment]
* We might have to add a hierarchy to the column names because multiple tables contain same column names

**Comment by lauradmartens** [comment]
* Check why inner join `tb1.left_join(tb2, 'gene_id').execute()` returns one column called `gene_id` while left join returns `gene_id_x` and `gene_id_y`

**Comment by lauradmartens** [comment]
This test needs to be checked: https://github.com/scverse/genomic-features/blob/45d97ac1d586c7c11bfa54afb8e1445ef1e5cab6/tests/test_columns.py#L79

File: src/genomic_features/ensembl/ensembldb.py [hunk 1]
```diff
@@ -115,15 +116,214 @@ def __init__(self, connection: ibis.BaseBackend):
...
+        query = query.filter(filter.convert()).select(cols)
+        return query
+
+    def join_query(self, tables, start_with, join_type="inner"):
+        """Join tables and return a query."""
+        # check for intermediate tables
+
+        tables.remove(start_with)
+        tables = [start_with] + tables
+        print(tables)
```
Comment (ivirshup) [inline src/genomic_features/ensembl/ensembldb.py:166]:
```suggestion
```

My bad

File: src/genomic_features/ensembl/ensembldb.py [hunk 2]
```diff
@@ -115,15 +116,214 @@ def __init__(self, connection: ibis.BaseBackend):
...
+
+    def tables_by_degree(self, tab: list[str] = None) -> list:
+        """Order tables available in the genomic features database."""
+        if tab is None:
+            tab = self.list_tables()  # list of table names
+        # check that all tables are in the database and print warning
+        if not set(tab).issubset(set(self.list_tables())):
+            print(
+                f"Warning: tables {set(tab) - set(self.list_tables())} are not in the database."
+            )
```
Comment (ivirshup) [inline src/genomic_features/ensembl/ensembldb.py:217]: Should this be an error?

If it should be a warning, how about [`warnings.warn`](https://docs.python.org/3/library/warnings.html#warnings.warn) instead?

File: tests/test_columns.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    assert result.shape == (185904, 4)
+    assert list(result.columns) == [
+        "gene_id",
+        "gene_name",
+        "protein_id",
+        "gene_biotype",
+    ]
+    assert (
+        result.loc[result.gene_biotype != "protein_coding", "protein_id"].isna().all()
+    )
```
Comment (ivirshup) [inline tests/test_columns.py:112]: Does this work now?

**Review by ivirshup — COMMENTED** [review]
Looks good! A few minor things, and one major thing:

Does joining on the protein table work now? That would be great!

File: tests/test_columns.py (hunk 3 above)
Comment (lauradmartens) [reply tests/test_columns.py:112]: Yes, this works now. I added the mapping function to ensembldb.py

File: src/genomic_features/ensembl/ensembldb.py (hunk 2 above)
Comment (lauradmartens) [reply src/genomic_features/ensembl/ensembldb.py:217]: At the moment it automatically deletes the tables that are not in the database and prints a warning (which I can replace by `warnings.warn`) unless you think error is preferable

**Review by ivirshup — APPROVED** [review]

**Pull request merged**